src/Altinn.App.Core/Internal/App/AppMetadata.cs

```diff
@@ -0,0 +1,112 @@
+using System.Text;
+using System.Text.Json;
+using Altinn.App.Core.Configuration;
+using Altinn.App.Core.Models;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+
+namespace Altinn.App.Core.Internal.App
+{
+    /// <summary>
+    /// Default implementation of IAppMetadata
+    /// </summary>
+    public class AppMetadata : IAppMetadata
+    {
+        private readonly ILogger<AppMetadata> _logger;
+        private readonly AppSettings _settings;
+        private readonly IFrontendFeatures _frontendFeatures;
+        private ApplicationMetadata? _application;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="AppMetadata"/> class.
+        /// </summary>
+        /// <param name="settings">The app repository settings.</param>
+        /// <param name="frontendFeatures">Application features service</param>
+        /// <param name="logger">A logger from the built in logger factory.</param>
+        public AppMetadata(
+            IOptions<AppSettings> settings,
+            IFrontendFeatures frontendFeatures,
+            ILogger<AppMetadata> logger)
+        {
+            _settings = settings.Value;
+            _frontendFeatures = frontendFeatures;
+            _logger = logger;
+        }
+
+        /// <inheritdoc />
+        /// <exception cref="System.Text.Json.JsonException">Thrown if deserialization fails</exception>
+        /// <exception cref="System.IO.FileNotFoundException">Thrown if applicationmetadata.json file not found</exception>
+        public async Task<ApplicationMetadata> GetApplicationMetadata()
+        {
+            // Cache application metadata
+            if (_application != null)
+            {
+                return _application;
+            }
+
+            string filename = _settings.AppBasePath + _settings.ConfigurationFolder + _settings.ApplicationMetadataFileName;
+            try
+            {
+                if (File.Exists(filename))
+                {
+                    JsonSerializerOptions jsonSerializerOptions = new JsonSerializerOptions()
```

Comment:
Ideally we should have these as default settings thus avoiding having to repeat them, but I'm not sure if it's possible. Just something we could keep in mind if we end up repeating these all over.

Reply:
Agreed, have removed the generic exception part as I didn't affect anything authorisation wise